client/src/quic_client.rs

```diff
@@ -173,76 +171,65 @@ impl TpuConnection for QuicTpuConnection {
             connection_stats,
         }
     }
+}
 
+impl TpuConnection for QuicTpuConnection {
     fn tpu_addr(&self) -> &SocketAddr {
         &self.client.addr
     }
 
-    fn send_wire_transaction<T>(
-        &self,
-        wire_transaction: T,
-        stats: &ClientStats,
-    ) -> TransportResult<()>
-    where
-        T: AsRef<[u8]>,
-    {
-        let _guard = RUNTIME.enter();
-        let send_buffer =
-            self.client
-                .send_buffer(wire_transaction, stats, self.connection_stats.clone());
-        RUNTIME.block_on(send_buffer)?;
-        Ok(())
-    }
-
-    fn send_wire_transaction_batch<T>(
-        &self,
-        buffers: &[T],
-        stats: &ClientStats,
-    ) -> TransportResult<()>
+    fn send_wire_transaction_batch<T>(&self, buffers: &[T]) -> TransportResult<()>
     where
         T: AsRef<[u8]>,
     {
+        let stats = ClientStats::default();
+        let len = buffers.len();
         let _guard = RUNTIME.enter();
         let send_batch = self
             .client
-            .send_batch(buffers, stats, self.connection_stats.clone());
-        RUNTIME.block_on(send_batch)?;
+            .send_batch(buffers, &stats, self.connection_stats.clone());
+        let res = RUNTIME.block_on(send_batch);
+        self.connection_stats
+            .add_client_stats(&stats, len, res.is_ok());
+        res?;
         Ok(())
     }
 
-    fn send_wire_transaction_async(
-        &self,
-        wire_transaction: Vec<u8>,
-        stats: Arc<ClientStats>,
-    ) -> TransportResult<()> {
+    fn send_wire_transaction_async(&self, wire_transaction: Vec<u8>) -> TransportResult<()> {
+        let stats = Arc::new(ClientStats::default());
```

Comment:
nit: doesn't seem like these really need to be an `Arc`?  could just declare the temporary inside the async closure and pass a mut ref to the client methods?

not a change for this PR, but consider a followup